scale-tests/deploy-batch-marathon-app.py

```diff
@@ -48,7 +48,7 @@ def _get_app_defn(app_id: str, dcos_username: str, dcos_password: str, input_fil
         """
         app_defn = {
             "id": app_id,
-            "cmd": "cd $MESOS_SANDBOX; git clone https://github.com/mesosphere/spark-build.git; cd spark-build/; pwd; git checkout $SPARK_BUILD_BRANCH; python3 -m venv test-env; pip3 install -r scale-tests/requirements.txt; dcos cluster setup https://master.mesos --username=$DCOS_UID --password=$DCOS_PASSWORD --no-check; dcos package install spark --cli --yes; cd scale-tests/; export PYTHONPATH=../spark-testing:../testing; python3 batch_test.py $MESOS_SANDBOX/$SCRIPT_ARGS",
+            "cmd": "(while sleep 600; do echo 'Refreshing DC/OS CLI authentication (interval: 600s)'; dcos auth login --username=${DCOS_UID} --password=${DCOS_PASSWORD}; done &); cd $MESOS_SANDBOX; git clone https://github.com/mesosphere/spark-build.git; cd spark-build/; pwd; git checkout $SPARK_BUILD_BRANCH; python3 -m venv test-env; pip3 install -r scale-tests/requirements.txt; dcos cluster setup https://master.mesos --username=$DCOS_UID --password=$DCOS_PASSWORD --no-check; dcos package install spark --cli --yes; cd scale-tests/; export PYTHONPATH=../spark-testing:../testing; python3 batch_test.py $MESOS_SANDBOX/$SCRIPT_ARGS",
```

Comment:
Meaning that it would also work for the GPU jobs?

Reply:
Having it in the Marathon app deployer is arguably better since it will work for all job scripts.

Currently our actual Python job scripts are unfortunately in [personal S3 buckets](https://github.com/mesosphere/spark-build/blob/9e18b9f33d4d7e92e3135b61ed8018811ddeba3e/scale-tests/batch_test.py#L56-L57). I've created [DCOS-43667](https://jira.mesosphere.com/browse/DCOS-43667) to address that. Maybe we can transplant the auth refresh to them in a better way after that.